addons/web/static/src/views/kanban/kanban_header.js

```diff
@@ -104,13 +104,10 @@ export class KanbanHeader extends Component {
         let name = displayName;
         if (groupByField.type === "boolean") {
             name = name ? _t("Yes") : _t("No");
+        } else if (groupByField.type === "integer") {
+            name = displayName === false ? "0" : displayName;
         } else if (!name) {
-            if (
-                isRelational(groupByField) ||
-                groupByField.type === "date" ||
-                groupByField.type === "datetime" ||
-                isNull(name)
-            ) {
+            if (isRelational(groupByField) || displayName === false || isNull(name)) {
```

Comment:
I would go for the same logic as in list which is simpler and handles the exact same usecase:
```
if boolean: yes or no
elif integer: displayname or 0
else: displayname or None
```